### PR TITLE
Update documentation for Vitis 2020.1 and Ubuntu 20.04

### PR DESCRIPTION
**Pull request by keryell**
This is a major update to latest version of Intel DPC++ SYCL to latest Vitis for Xilinx FPGA.
@j-stephan if you have some spare time to try this new version on your system, I am interested.
We are looking for some people outside of Xilinx to polish the install process...
There are still some issues, but in a few days it should be OK to try.

### CODE REVIEW

**Comment by j-stephan** [comment]
This seems to work quite nicely. On Ubuntu 20.04 I took the following steps:

* Install Vitis 2020.01
* Build and install latest XRT `master` and the deployment + development shell
* Clone latest `sycl/unified/next`
* Run `configure.py` and `compile.py`
* Set up all the environment variables as described in the "Getting started with Alveo" document.

I ran into some smaller issues:

* During compilation XRT picked up `libOpenCL.so.1` from my local CUDA installation, not the system version. This gave me some headaches because it refused to launch the software emulation. LD_LIBRARY_PATH wasn't set to find CUDA so this may have been some issue with the ICD loader. I uninstalled CUDA for now.

* `compile.py` failed because the git tag for the level zero loader doesn't exist anymore. I had to manually revert it to an older tag (newer tags apparently introduced an API breaking change).

* `sw_emu` compiles and runs. `hw_emu` fails in link stage:

```
****** vpl v2020.1 (64-bit)
  **** SW Build 2902540 on Wed May 27 19:54:35 MDT 2020
    ** Copyright 1986-2020 Xilinx, Inc. All Rights Reserved.

INFO: [VPL 60-839] Read in kernel information from file '/tmp/sycl-xocc.7tuM86_parallel_for_ND_range_linker/_x/link/int/kernel_info.dat'.
INFO: [VPL 60-423]   Target device: xilinx_u200_xdma_201830_2
INFO: [VPL 60-1032] Extracting hardware platform to /tmp/sycl-xocc.7tuM86_parallel_for_ND_range_linker/_x/link/vivado/vpl/.local/hw_platform
[14:09:48] Run vpl: Step create_project: Started
Creating Vivado project.
[14:09:49] Run vpl: Step create_project: Completed
[14:09:49] Run vpl: Step create_bd: Started
[14:10:30] Run vpl: Step create_bd: Completed
[14:10:30] Run vpl: Step update_bd: Started
[14:10:33] Run vpl: Step update_bd: Completed
[14:10:33] Run vpl: Step generate_target: Started
[14:10:56] Run vpl: Step generate_target: Completed
[14:10:56] Run vpl: Step config_hw_emulation: Started
[14:11:46] Run vpl: Step config_hw_emulation: Failed
[14:11:49] Run vpl: FINISHED. Run Status: config_hw_emulation ERROR
ERROR: [VPL 60-773] In '/tmp/sycl-xocc.7tuM86_parallel_for_ND_range_linker/_x/link/vivado/vpl/runme.log', caught Tcl error:  ./elaborate.sh: Zeile 20: LIBRARY_PATH ist nicht gesetzt.
ERROR: [VPL 60-1328] Vpl run 'vpl' failed
ERROR: [VPL 60-806] Failed to finish platform linker
INFO: [v++ 60-1442] [14:11:49] Run run_link: Step vpl: Failed
Time (s): cpu = 00:06:07 ; elapsed = 00:02:22 . Memory (MB): peak = 1332.305 ; gain = 0.000 ; free physical = 38535 ; free virtual = 57091
ERROR: [v++ 60-661] v++ link run 'run_link' failed
ERROR: [v++ 60-626] Kernel link failed to complete
ERROR: [v++ 60-703] Failed to finish linking
INFO: [v++ 60-1653] Closing dispatch client.
clang-12: error: sycl-link-xocc command failed with exit code 1 (use -v to see invocation)
```

The error message translates to `Line 20: LIBRARY_PATH is not set.` Is there an environment variable I need to set somewhere?

File: sycl/doc/GettingStartedAlveo.md [hunk 1]
```diff
@@ -283,16 +290,15 @@ Some checks may fail but that may not be an issue.
 ## Compiling and running an application
 
 The typical environment is setup with something like
+FIXME
 ```bash
-export XILINX_SDX=/opt/xilinx/SDx/2019.1
-export XILINX_VIVADO=/opt/xilinx/Vivado/2019.1
-PATH=$PATH:/opt/xilinx/xrt/bin:$XILINX_SDX/bin:$XILINX_VIVADO/bin
-# This is the platform of the shell on the FPGA board
-export XILINX_PLATFORM=xilinx_u200_xdma_201830_1
```
Comment (Ralender) [inline sycl/doc/GettingStartedAlveo.md:294]: I also worked a improving documentation in this file.
very weirdly at least one of the test doesn't seem to terminate 
when XILINX_SDX isn't set so we still need it.
also for at least hw_emu LIBRARY_PATH needs to be set.

File: sycl/doc/Tests.md [hunk 2]
```diff
@@ -19,8 +19,8 @@ working yet are in the issue_related directory.
 All the tests in the simple_tests directory are small unit tests with the
```
Comment (Ralender) [inline sycl/doc/Tests.md]: Since the testing has been significantly changed so i expect that there is still quite a few things to change in this file,

**Review by Ralender — CHANGES_REQUESTED** [review]
I think this is overall a good update.
there is still a few thing to update though, I also update the document in one of my commit on my Merging2 branch so we are going to have conflicts.

**Comment by Ralender** [comment]
> This seems to work quite nicely. On Ubuntu 20.04 I took the following steps:
> 
> * Install Vitis 2020.01
> * Build and install latest XRT `master` and the deployment + development shell
> * Clone latest `sycl/unified/next`
> * Run `configure.py` and `compile.py`
> * Set up all the environment variables as described in the "Getting started with Alveo" document.
> 
> I ran into some smaller issues:
> 
> * During compilation XRT picked up `libOpenCL.so.1` from my local CUDA installation, not the system version. This gave me some headaches because it refused to launch the software emulation. LD_LIBRARY_PATH wasn't set to find CUDA so this may have been some issue with the ICD loader. I uninstalled CUDA for now.

I have not seen this issue 
> * `compile.py` failed because the git tag for the level zero loader doesn't exist anymore. I had to manually revert it to an older tag (newer tags apparently introduced an API breaking change).

This has been fixed on my development branch at https://github.com/Ralender/sycl/tree/Merging2 if you want to try.
> * `sw_emu` compiles and runs. `hw_emu` fails in link stage:
> 
> ```
> ****** vpl v2020.1 (64-bit)
>   **** SW Build 2902540 on Wed May 27 19:54:35 MDT 2020
>     ** Copyright 1986-2020 Xilinx, Inc. All Rights Reserved.
> 
> INFO: [VPL 60-839] Read in kernel information from file '/tmp/sycl-xocc.7tuM86_parallel_for_ND_range_linker/_x/link/int/kernel_info.dat'.
> INFO: [VPL 60-423]   Target device: xilinx_u200_xdma_201830_2
> INFO: [VPL 60-1032] Extracting hardware platform to /tmp/sycl-xocc.7tuM86_parallel_for_ND_range_linker/_x/link/vivado/vpl/.local/hw_platform
> [14:09:48] Run vpl: Step create_project: Started
> Creating Vivado project.
> [14:09:49] Run vpl: Step create_project: Completed
> [14:09:49] Run vpl: Step create_bd: Started
> [14:10:30] Run vpl: Step create_bd: Completed
> [14:10:30] Run vpl: Step update_bd: Started
> [14:10:33] Run vpl: Step update_bd: Completed
> [14:10:33] Run vpl: Step generate_target: Started
> [14:10:56] Run vpl: Step generate_target: Completed
> [14:10:56] Run vpl: Step config_hw_emulation: Started
> [14:11:46] Run vpl: Step config_hw_emulation: Failed
> [14:11:49] Run vpl: FINISHED. Run Status: config_hw_emulation ERROR
> ERROR: [VPL 60-773] In '/tmp/sycl-xocc.7tuM86_parallel_for_ND_range_linker/_x/link/vivado/vpl/runme.log', caught Tcl error:  ./elaborate.sh: Zeile 20: LIBRARY_PATH ist nicht gesetzt.
> ERROR: [VPL 60-1328] Vpl run 'vpl' failed
> ERROR: [VPL 60-806] Failed to finish platform linker
> INFO: [v++ 60-1442] [14:11:49] Run run_link: Step vpl: Failed
> Time (s): cpu = 00:06:07 ; elapsed = 00:02:22 . Memory (MB): peak = 1332.305 ; gain = 0.000 ; free physical = 38535 ; free virtual = 57091
> ERROR: [v++ 60-661] v++ link run 'run_link' failed
> ERROR: [v++ 60-626] Kernel link failed to complete
> ERROR: [v++ 60-703] Failed to finish linking
> INFO: [v++ 60-1653] Closing dispatch client.
> clang-12: error: sycl-link-xocc command failed with exit code 1 (use -v to see invocation)
> ```
> 
> The error message translates to `Line 20: LIBRARY_PATH is not set.` Is there an environment variable I need to set somewhere?

yes LIBRARY_PATH needs to be set to the standard search path for libraries so on my system it is 
export LIBRARY_PATH=/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/

**Comment by j-stephan** [comment]
Setting `LIBRARY_PATH` to `/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/` fixed the problem. I am now able to compile and execute in `hw_emu` mode. Thanks!

**Comment by keryell** [comment]
Thanks for the feedback, guys!
@j-stephan by  "There are still some issues, but in a few days it should be OK to try." I was not expecting you to test it already, since I knew there were still some issues on our side, but about booking some of your time in the near future when it seems ready... :-)
But thanks for having tried it anyway!

File: sycl/doc/GettingStartedAlveo.md (hunk 1 above)
Comment (keryell) [reply sycl/doc/GettingStartedAlveo.md:294]: OK, keeping it and I will investigate later.

File: sycl/doc/Tests.md (hunk 2 above)
Comment (keryell) [reply sycl/doc/Tests.md]: I have updated most of them.

**Pull request merged**